### ISSUE TITLE
COMPOSITE_SOFTLIGHT v7 issue

### ISSUE THREAD

**Issue by myrecs** [Imagick/imagick#485]
`$canvas = new Imagick('test.png'); //image with transparent pieces
$gradient = new Imagick();
$gradient->newPseudoImage($canvas->getImageWidth(), $canvas->getImageHeight(), 'gradient:#979797-#373737');
$gradient->compositeImage($canvas, imagick::COMPOSITE_COPYOPACITY , 0, 0);
$canvas->compositeImage($gradient, imagick::COMPOSITE_SOFTLIGHT, 0, 0);
  
$im = new Imagick('base.jpg');
$im->compositeImage($canvas, imagick::COMPOSITE_OVER, 0, 0);`

another one issue in v7 imagick. installed win 7.0.7-11 
COMPOSITE_SOFTLIGHT const produces non transparent (black) result on composing, when it was used `in` upper layer before composing. other consts work well.
v6 works well

**Comment by myrecs** [comment]
![test](https://user-images.githubusercontent.com/32914184/139538813-8022c6d1-13c6-4511-b752-f6d1b5c17637.png)
![base](https://user-images.githubusercontent.com/32914184/139538816-a83096c9-17b0-4b68-8d7b-5f9dded852c7.png)

**Comment by myrecs** [comment]
moreover, what is interesting is that before final composing $canvas holds good transparent png (of course i dont know about it's normality under the hood), but if to save it as png before composing step it will be png with transparency. 
If then take this png as top layer with "new Imagick('path_to_that_png')" instantiation and make composing everything will be fine.

**Comment by Danack** [comment]
What versions of ImageMagick do you have on your system? I just tried your code with:

ImageMagick 6.9.12-27 Q16 x86_64 2021-10-24
ImageMagick 7.1.0-4 Q16 x86_64 2021-07-18

And I seem to get the same result for both.
![445_bug](https://user-images.githubusercontent.com/1505719/139590498-92c608c1-5f08-4792-9678-f5c3116b7591.jpg)

If you use docker, you can run https://github.com/Imagick/ImagickDemos to have the example site come up, and it has both ImageMagick 6 and 7 available on it, which makes it easy to compare.

The code for your bug report is at the end of src/ImagickDemo/Imagick/functions.php . I haven't put the images into VCS, so you would need to save them as 485_car.png and 485_base.jpg into the directory src/ImagickDemo/Imagick

**Comment by myrecs** [comment]
i pointed it in my first message
ImageMagick 7.0.7-11 Q16 x64 2017-11-23

**Comment by Danack** [comment]
I've asked someone to update the version of ImageMagick that is used to build Imagick. Technically you can get ImageMagick itself at https://imagemagick.org/script/download.php#windows, but that would require you to build Imagick against that, which to be fair not many windows users are used to.

I see the same behaviour you see on that version of ImageMagick, so presumably it is a bug in that.

**Comment by myrecs** [comment]
this is a very problematic version of the program, so it's surprising that no one has yet tried to get rid of it

**Comment by myrecs** [comment]
Please if you have someone who could make and rebuild imagick with the latest version for win and php 7.4, ask  him to update the repo! thx!

**Comment by Danack** [comment]
Yeah, CMB is working on it, and making it be automated, so that it doesn't take so long in future.

**Comment by Danack** [comment]
@myrecs Hi - are you able to test the newly built versions from https://windows.php.net/downloads/pecl/snaps/imagick/3.5.1/ ?

The way it's built is slightly different so there may be un-intended changes. Or there might not be. I'll open a new issue for collating problems.

**Comment by myrecs** [comment]
@Danack SOFT_LIGHT seems to be good, but new issue with COMPOSITE_MULTIPLY
```
$canvas = new Imagick('top.png');
$im = new Imagick('bottom.png');
$im->compositeImage($canvas, imagick::COMPOSITE_MULTIPLY, 0, 0);
$im->writeImage('res.png');
```
this wasn't earlier

![top](https://user-images.githubusercontent.com/32914184/140589807-358a966b-bd5b-4bff-9a1c-ba99ca55e761.png)
![res](https://user-images.githubusercontent.com/32914184/140589808-dffc9c9e-e96d-4411-a8e1-7a1e9b766022.png)
![bottom](https://user-images.githubusercontent.com/32914184/140589809-a79071fe-f6d3-4f21-8cda-d5259bfdda89.png)

**Comment by Danack** [comment]
Are the first two images 'top.png' and 'bottom.png', or is the photo of the beatles one of them?

**Comment by myrecs** [comment]
@Danack first-top layer, beatles-bottom, in center-strange result

**Comment by Danack** [comment]
Looks like a bug in ImageMagick itself, as it seems to be the same on the command line. Have opened an issue upstream: https://github.com/ImageMagick/ImageMagick/issues/4429
